web/embedding.py

```diff
@@ -238,7 +247,27 @@ def _from_word2vec_binary(fname):
             elif len(words) > vocab_size:
                 raise RuntimeError("Read too many words, incorrect file")
 
-            return words, vectors
+            return Embedding._deduplicate(words, vectors)
+
+    @staticmethod
+    def _deduplicate(words, vectors):
+        """deduplicate words with triming"""
+
+        word_dict = dict(zip(words, vectors))
+
+        cleaned_words = dict([])
+        for w in word_dict.keys():
+
+            c_w = w.strip()
```

Comment:
Miała być logika, że preferujemy słowo, które po zestripowaniu jest takie samo. Teraz Twoj kod jak ma "cow ", oraz "cow", a "cow " jest po "cow" to nadpisze nam wektor :(